### PR TITLE
Feature/share screen

### PR DESCRIPTION
**Pull request by hasevr**
Screen share by main screen.
Also mute is fixed.

### CODE REVIEW

**Comment by hasevr** [comment]
Still small problem (use better logger) remains, desktop sharing for main screen now works.
For audio track, need a bit more work.

**Comment by hasevr** [comment]
Fix mute unmute problem video, or  #48. It was cased by modification of lib-jitsi-meet.

File: services/conference/src/scripts/components/map/BackgroundLayer/Background.tsx [hunk 1]
```diff
@@ -6,29 +6,55 @@ import React from 'react'
...
-    backgroundColor: '#DFDBE5',
-    backgroundImage: `url("data:image/svg+xml,%3Csvg width='80' height='80' viewBox='0 0 80 80' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='none' fill-rule='evenodd'%3E%3Cg fill='%239C92AC' fill-opacity='0.4'%3E%3Cpath d='M50 50c0-5.523 4.477-10 10-10s10 4.477 10 10-4.477 10-10 10c0 5.523-4.477 10-10 10s-10-4.477-10-10 4.477-10 10-10zM10 10c0-5.523 4.477-10 10-10s10 4.477 10 10-4.477 10-10 10c0 5.523-4.477 10-10 10S0 25.523 0 20s4.477-10 10-10zm10 8c4.418 0 8-3.582 8-8s-3.582-8-8-8-8 3.582-8 8 3.582 8 8 8zm40 40c4.418 0 8-3.582 8-8s-3.582-8-8-8-8 3.582-8 8 3.582 8 8 8z' /%3E%3C/g%3E%3C/g%3E%3C/svg%3E")`,
-    height: SIZE,
-    width: SIZE,
+  img: (props:BackgroundProps) => {
+    if (props.isTransparnet) {
+      return  {
+        position: 'absolute',
+        transform: `translate(-${SIZE / 2}px, -${SIZE / 2}px)`,
+        backgroundColor: 'rgba(0,0,0,0.1)',
```
Comment (zlguo1996) [inline services/conference/src/scripts/components/map/BackgroundLayer/Background.tsx:14]: It seems the return objects are almost same, why not use `backgroundColor: props.isTransparent ? xxx : xxx`

File: services/conference/src/scripts/components/map/map.tsx [hunk 2]
```diff
@@ -2,16 +2,20 @@ import {Base} from '@components/map/Base'
...
 export const Map: React.FC<BaseProps> = (props) => {
+  const store = useContentsStore()
+  const stream = useObserver(() => store.mainStream)
+
   return (
     <Base {...props}>
-      <BackgroundLayer />
-      <ShareLayer />
+      {<BackgroundLayer isTransparnet={stream !== undefined} />}
+      {stream ? <div /> : <ShareLayer />}
```
Comment (zlguo1996) [inline services/conference/src/scripts/components/map/map.tsx:18]: `{stream ? undefined : <ShareLayer />}` seems better.

File: services/conference/src/scripts/models/audio/StereoManager.ts [hunk 3]
```diff
@@ -87,10 +89,8 @@ export class StereoManager {
 
   set audioOutputMuted(muted: boolean) {
     console.log('audioOutputMuted', muted)
-    if (muted) {
-      this.audioElement.pause()
-    }else {
-      this.audioElement.play()
+    for (const id in this.nodes) {
+      this.nodes[id].setMute(muted)
```
Comment (zlguo1996) [inline services/conference/src/scripts/models/audio/StereoManager.ts:93]: This setting seems have conflict with individual mute.
For example, we mute our speaker, but in individual A's setting, we unmute A. Then our speaker would play sound from A. That's unexpected behavior.
A better solution would have two settings: global mute and individual mute. We use observe those two settings and change setting of NodeGroup for every individual.

Comment (hasevr) [reply services/conference/src/scripts/models/audio/StereoManager.ts:93]: This mute is for global mute. When footer's button clicked.
For individual mute remote different system is needed.

Comment (hasevr) [reply services/conference/src/scripts/models/audio/StereoManager.ts:93]: This mute is for global mute. When footer's button clicked.
For individual mute remote different system is needed.

Comment (hasevr) [reply services/conference/src/scripts/models/audio/StereoManager.ts:93]: This mute is for global mute. When footer's button clicked.
For individual mute remote different system is needed.

Comment (hasevr) [reply services/conference/src/scripts/models/audio/StereoManager.ts:93]: This mute is for global mute. When footer's button clicked.
For individual mute remote different system is needed.

Comment (zlguo1996) [reply services/conference/src/scripts/models/audio/StereoManager.ts:93]: Yeah, I understand. But when user only want to mute one participant, he have to call someNode.setMute(true).
For example, 
If user 1. mute global, 2. unmute participant A => participant A's sound would be played.
If user 1. unmute participant A, 2. mute global => participant A's sound would not be played.
That's unexpected and unconsistent.
The logic here should be: `A is muted = use mute global || user mute A`

Thus, the mute global API here in StereoManager is just used for stop the global audio output. If you want to mute individual stream, please use api in every NodeGroup.

File: services/conference/src/scripts/components/map/BackgroundLayer/Background.tsx (hunk 1 above)
Comment (hasevr) [reply services/conference/src/scripts/components/map/BackgroundLayer/Background.tsx:14]: The best appearance is still not obvious.
It woll change more.
SVG is also a bit different.

Comment (zlguo1996) [reply services/conference/src/scripts/components/map/BackgroundLayer/Background.tsx:14]: OK

File: services/conference/src/scripts/components/map/map.tsx (hunk 2 above)
Comment (hasevr) [reply services/conference/src/scripts/components/map/map.tsx:18]: I found undefined causes compile error.

Comment (hasevr) [reply services/conference/src/scripts/components/map/map.tsx:18]: So I have to keep <div />. Is there a better solution ?

File: services/conference/src/scripts/models/audio/StereoManager.ts (hunk 3 above)
Comment (hasevr) [reply services/conference/src/scripts/models/audio/StereoManager.ts:93]: I think mute each participant will be not needed. Too complex and bring many problems to users.

File: services/conference/src/scripts/components/map/map.tsx (hunk 2 above)
Comment (zlguo1996) [reply services/conference/src/scripts/components/map/map.tsx]: Compile error is caused by child type in BaseProps. I've fixed, please refer to: e387f35d0fde75190722b2308bb417cb0745f743

**Comment by hasevr** [comment]
I understand the reason of the error and confirm modification.
Thanks.

**Pull request merged**